lib/ProductOpener/Products.pm

```diff
@@ -1469,13 +1470,30 @@ sub store_product ($user_id, $product_ref, $comment) {
 	store("$new_data_root/products/$path/$rev.sto", $product_ref);
 
 	# Also store the product in MongoDB, unless it was marked as deleted
+	# Assuming you're within the store_product function and the product is marked as deleted
 	if ($product_ref->{deleted}) {
+    # Check if there are any images associated with the product
+		if (defined $product_ref->{images}) {
+			my $move_to = 'trash';  # Define the destination for deletion
+
+			# Iterate over each image ID stored in the product's images hash
+			foreach my $imgid (keys %{$product_ref->{images}}) {
+				# Ensure the image ID is numeric or follows your expected format
+				if ($imgid =~ /^\d+$/ || $imgid =~ /^(front|ingredients|nutrition)/) {
+					# Call the process_image_move function using each image ID
+					process_image_move($user_id, $product_ref->{code}, $imgid, $move_to, $product_ref->{owner});
```

Comment:
I would do this in an eval and eventually log an error, because it's important to avoid failing.
(eval used as a form of try / catch in perl).